hydro_lang/src/builder/deploy.rs

```diff
@@ -24,7 +25,7 @@ use crate::location::{Cluster, ExternalProcess, Location, LocationId, Process};
 use crate::staging_util::Invariant;
 
 pub struct DeployFlow<'a, D: LocalDeploy<'a>> {
-    pub(super) ir: Vec<HydroLeaf>,
+    pub(super) ir: UnsafeCell<Vec<HydroLeaf>>,
```

Comment:
What is the purpose of the `UnsafeCell`? Why not just have the owned value?

Reply:
In `preview_compile`, we want to take an `&self`, but `build_inner` requires an `&mut Vec<HydroLeaf>` because the shared logic for traversal under the hood operates on `&mut` (even though when emitting DFIR we don't ever mutate the IR).

Reply:
Why not just make `preview_compile` take `&self`?

Reply:
So `preview_compile` does not need exclusive access, since it doesn't do any actual mutations, so it's nice to just take an `&self` so that user code doesn't have to make their `DeployFlow` `mut` (it's also confusing the the user if `preview_compile` takes a mutable pointer because the point of the API is to compile without touching the IR or having any side effects).

Reply:
I could just replace this with a `RefCell`...

Reply:
I feel like using `UnsafeCell` is uglier and harder to maintain than just taking `&mut self` and leaving note in the docs, but either way. Just add a comment to this line to explain why UnsafeCell is needed

Reply:
Added a comment; I think the unsafety-ness is sufficiently scoped to not cause too much trouble.